pkg/operation/botanist/secrets.go

```diff
@@ -619,9 +627,62 @@ func (b *Botanist) generateWantedSecrets(basicAuthAPIServer *secrets.BasicAuth,
 				CertType:  secrets.ServerCert,
 				SigningCA: certificateAuthorities[gardencorev1alpha1.SecretNameCACluster],
 			},
+			// Secret for elasticsearch
+			&secrets.CertificateSecretConfig{
+				Name: "elasticsearch-logging-server",
+
+				CommonName:   "elasticsearch",
+				Organization: nil,
+				DNSNames:     elasticsearchHosts,
+				IPAddresses:  nil,
+
+				CertType:  secrets.ServerClientCert,
+				SigningCA: certificateAuthorities[gardencorev1alpha1.SecretNameCACluster],
+				PKCS:      secrets.PKCS8,
+			},
 			// Secret definition for logging
-			&secrets.BasicAuthSecretConfig{
-				Name:   "logging-ingress-credentials",
+			&secrets.BasicAuthHashSecretConfig{
+				Name:   "logging-kibana-user-credentials",
+				Format: secrets.BasicAuthFormatNormal,
+
+				Username:       "user",
+				PasswordLength: 32,
+			},
+			&secrets.BasicAuthHashSecretConfig{
+				Name:   "logging-kibana-admin-credentials",
+				Format: secrets.BasicAuthFormatNormal,
+
+				Username:       "admin",
+				PasswordLength: 32,
+			},
+			&secrets.CertificateSecretConfig{
+				Name: "sg-admin-client",
+
+				CommonName:   "elasticsearch-logging",
+				Organization: nil,
+				DNSNames:     sgHosts,
+				IPAddresses:  nil,
+
+				CertType:  secrets.ClientCert,
+				SigningCA: certificateAuthorities[gardencorev1alpha1.SecretNameCACluster],
+				PKCS:      secrets.PKCS8,
```

Comment:
Why is this needed?

Reply:
Because search guard works only with PKCS8 encryption format. Gardener had only PKCS1, so I extended it, and now we can choose between 2 formats.

Reply:
OK, thank's for clarifying!